src/main/java/com/yahoo/bullet/operations/aggregations/sketches/Sketch.java

```diff
@@ -97,14 +95,14 @@ public Clip getResult(String metaKey, Map<String, String> conceptKeys) {
     protected abstract Integer getSize();
 
     /**
-     * Utility function to add a key to the metadata if the key is not null.
+     * Utility function to add a key to the metadata if the key and value are not null.
      *
      * @param metadata The non-null {@link Map} representing the metadata.
      * @param key The key to add if not null.
      * @param supplier A {@link Supplier} that can produce a value to add to the metadata for the key. If the supplier
      *                 produces null, it is not added.
      */
-    static void addIfKeyNonNull(Map<String, Object> metadata, String key, Supplier<Object> supplier) {
+    static void addIfNonNull(Map<String, Object> metadata, String key, Supplier<Object> supplier) {
         Object data = null;
         if (key != null) {
             data = supplier.get();
```

Comment:
A few lines below this you do metadata.put(key, supplier.get())  -  want to do metadata.put(key, data) instead to save the extra function call?